src/go/rpk/pkg/cli/connect/install.go

```diff
@@ -0,0 +1,154 @@
+			out.MaybeDieErr(err)
+			_, installed := plugin.ListPlugins(fs, plugin.UserPaths()).Find("connect")
+			if installed && !force {
+				if version != "latest" {
+					out.Exit("Redpanda connect is already installed. Use --force to force installation, or delete current version with 'rpk connect uninstall' first")
+				}
+				out.Exit("Redpanda connect is already installed.\nIf you want to upgrade to the latest version, please run 'rpk connect upgrade'.")
+			}
+			_, err = installConnect(cmd.Context(), fs, version)
+			out.MaybeDie(err, "unable to install Redpanda Connect: %v; if running on an air-gapped environment you may install 'redpanda-connect' with your package manager.", err)
+
+			fmt.Println("Redpanda Connect successfully installed.")
+		},
+	}
+	cmd.Flags().BoolVar(&force, "force", false, "Force install of Redpanda Connect")
+	cmd.Flags().StringVar(&version, "connect-version", "latest", "Redpanda Connect version to install. (e.g. 4.32.0)")
+	return cmd
+}
+
+// installConnect installs Redpanda Connect plugin in the default location.
+// If the plugin is already installed, it will return early and won't download
+// the latest plugin. Version string let you select a specific version to
+// download, if "latest" or an empty string is passed, it will download the
+// latest version.
+func installConnect(ctx context.Context, fs afero.Fs, version string) (path string, err error) {
+	// We check this before calling the API, to avoid getting the binary
+	// and later fail due to a user setting.
+	pluginDir, err := plugin.DefaultBinPath()
+	if err != nil {
+		return "", fmt.Errorf("unable to determine plugin default path: %v", err)
+	}
+
+	art, _, err := getConnectArtifact(ctx, version)
+	if err != nil {
+		return "", err
+	}
+
+	return downloadAndInstallConnect(ctx, fs, pluginDir, art.Path, art.Sha256)
+}
+
+func getConnectArtifact(ctx context.Context, version string) (connectArtifact, string, error) {
+	plCl, err := newRepoClient()
+	if err != nil {
+		return connectArtifact{}, "", err
+	}
+	manifest, err := plCl.Manifest(ctx)
+	if err != nil {
+		return connectArtifact{}, "", err
+	}
+	var (
+		art        connectArtifact
+		retVersion string
+	)
+	if version == "latest" || version == "" {
+		art, retVersion, err = manifest.LatestArtifact()
+		if err != nil {
+			return connectArtifact{}, "", err
+		}
+	} else {
+		art, err = manifest.ArtifactVersion(version)
+		if err != nil {
+			return connectArtifact{}, "", err
+		}
+		retVersion = version
+	}
+	return art, retVersion, nil
+}
+
+func downloadAndInstallConnect(ctx context.Context, fs afero.Fs, installPath, downloadURL, expShaPrefix string) (string, error) {
+	bin, err := plugin.Download(ctx, downloadURL, true, expShaPrefix)
+	if err != nil {
+		return "", fmt.Errorf("unable to download Redpanda Connect from %q: %v", downloadURL, err)
+	}
+
+	// Ensure the dir exists, but if we have to create it, we do not want
+	// sudo.
+	if exists, _ := afero.DirExists(fs, installPath); !exists {
```

Comment:
swallowing error?

Reply:
No, this is intentional. If there is an error (fails to stat: https://github.com/spf13/afero/blob/master/util.go#L255) or it doesn't exists, we try to recreate it. If we fail, we send the error back.

We use the same approach in:

1. https://github.com/redpanda-data/redpanda/blob/cc778b9d57162684d41cc3ad02c9cfd389001d62/src/go/rpk/pkg/cli/plugin/install.go#L169
2. https://github.com/redpanda-data/redpanda/blob/cc778b9d57162684d41cc3ad02c9cfd389001d62/src/go/rpk/pkg/cli/cloud/byoc/install.go#L166

Reply:
It might be worth making that even more explicit in the comment 😅

Reply:
Added a better comment here :+1: